api/trace/sampler.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import "go.opentelemetry.io/api/core"
+
+type Sampler interface {
+	// ShouldSample returns a Decision that contains a decision whether to sample
+	// or not sample the span to be created. Decision is based on a Sampler specific
+	// algorithm that takes into account one or more input parameters.
+	ShouldSample(
+		sc core.SpanContext,
+		remote bool,
+		traceID core.TraceID,
+		spanID uint64,
+		spanName string,
+	) Decision
```

Comment:
Comments on the parameter names would help, here. For instance, I'm wondering about the relationship between `spanID` and `traceID` and `sc`. 

(I think there's going to be a discussion about sampling at the spec level, maybe we should hold off?)

Reply:
I think we should commit this so we get basic samplers (always and never) available for testing. There are bunch of [sampler related issues](https://github.com/open-telemetry/opentelemetry-specification/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+sampler). When those gets resolved we can update accordingly.